### PR TITLE
Improved test coverage for rdkit.Chem.Fingerprints

### PR DESCRIPTION
**Pull request by gedeck**
A few additional tests to improve the test coverage.

### CODE REVIEW

File: rdkit/Chem/Fingerprints/UnitTestSimScreener.py [hunk 1]
```diff
@@ -17,21 +17,58 @@ def fingerprinter(mol):
...
+
+    screener.Reset()
+
   def test1_TopNScreener(self):
     smis = ['C1CCCCC1', 'C1OCCCC1', 'C1NCCCC1', 'c1ccccc1', 'C1C(C)CCCC1', 'C1C(C)C(C)CCC1']
     suppl = Chem.SmilesMolSupplierFromText('\n'.join(smis), delimiter=",", smilesColumn=0,
                                            nameColumn=-1, titleLine=0)
     metric = DataStructs.TanimotoSimilarity
-    probe = fingerprinter(Chem.MolFromSmiles('C1OCCCC1'))
+    mol = Chem.MolFromSmiles('C1OCCCC1')
```
Comment (greglandrum) [inline rdkit/Chem/Fingerprints/UnitTestSimScreener.py]: Any particular reason for this change or is it just something you prefer/find more readable?

File: rdkit/Chem/Fingerprints/UnitTestSimScreener.py [hunk 2]
```diff
@@ -17,21 +17,58 @@ def fingerprinter(mol):
...
     matches2 = [x for x in screener]
     self.assertEqual(len(matches2), 3)
     self.assertEqual(matches1, matches2)
 
+    self.assertEqual(probe, screener.GetSingleFingerprint(mol))
+
+    # Getting the length also triggers the execution of the screen
+    screener = SimilarityScreener.TopNScreener(3, probe=probe, metric=metric,
+                                               fingerprinter=fingerprinter, dataSource=suppl)
+    self.assertEqual(screener.topN, None)
```
Comment (greglandrum) [inline rdkit/Chem/Fingerprints/UnitTestSimScreener.py]: Not important, but I think this should really be `self.assertIsNone()` and the one a couple lines down should be `self.assertIsNotNone()`

**Comment by greglandrum** [comment]
@gedeck: I am fine accepting this as is, so if you don't want to make the couple of small changes just say so and I'll go ahead and merge.

File: rdkit/Chem/Fingerprints/UnitTestSimScreener.py (hunk 1 above)
Comment (gedeck) [inline rdkit/Chem/Fingerprints/UnitTestSimScreener.py]: None really. Usually, I prefer avoiding unnecessary variables. Will change it.

Comment (gedeck) [inline rdkit/Chem/Fingerprints/UnitTestSimScreener.py]: This one is actually used a few lines further down:
```
self.assertEqual(probe, screener.GetSingleFingerprint(mol))
```

File: rdkit/Chem/Fingerprints/UnitTestSimScreener.py (hunk 2 above)
Comment (gedeck) [inline rdkit/Chem/Fingerprints/UnitTestSimScreener.py]: One thing learned today ;-)

**Pull request merged**